plugins/qtosgrave/qtosgviewer.cpp

```diff
@@ -193,18 +199,20 @@ QtOSGViewer::~QtOSGViewer()
         boost::mutex::scoped_lock lock(_mutexGUIFunctions);
 
         list<GUIThreadFunctionPtr>::iterator itmsg;
-        FORIT(itmsg, _listGUIFunctions) {
-            try {
-                (*itmsg)->Call();
-            }
-            catch(const boost::bad_weak_ptr& ex) {
-                // most likely viewer
-            }
-            catch(const std::exception& ex2) {
-                RAVELOG_WARN_FORMAT("unexpected exception in gui function: %s", ex2.what());
+        FOREACH(itpriority, _mapGUIFunctionLists) {
+            FORIT(itmsg, itpriority->second) {
+                try {
+                    (*itmsg)->Call();
```

Comment:
@Hsifnus I'm actually wondering why we need to keep `_mutexGUIFunctions` locked during the Call... perhaps we can get away with not..?

Reply:
Given the nature of where and when the destructor is typically called, perhaps going without the lock does makes sense.